File: elixir/rna-transcription/dna.exs
defmodule DNA do
  @transcriptions %{?A => ?U, ?C => ?G, ?G => ?C, ?T => ?A}

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> DNA.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna |> Enum.map(&(Map.get(@transcriptions, &1)))
  end
end

File: haskell/rna-transcription/DNA.hs
module DNA (toRNA) where

mapCGAT :: Char -> Char
mapCGAT 'C' = 'G'
mapCGAT 'G' = 'C'
mapCGAT 'A' = 'U'
mapCGAT 'T' = 'A'
mapCGAT  _  = ' '

toRNA :: String -> String
toRNA = map mapCGAT
